src/ripple/app/misc/impl/Transaction.cpp

```diff
@@ -134,13 +117,27 @@ Transaction::pointer Transaction::load(uint256 const& id, Application& app)
         *db << sql, soci::into (ledgerSeq), soci::into (status),
                 soci::into (sociRawTxnBlob, rti);
         if (!db->got_data () || rti != soci::i_ok)
+        {
+            error_code = rpcTXN_NOT_FOUND;
             return {};
+        }
 
         convert(sociRawTxnBlob, rawTxn);
     }
 
-    return Transaction::transactionFromSQLValidated (
-        ledgerSeq, status, rawTxn, app);
+    std::shared_ptr<Transaction> txn;
+    try
+    {
+        txn = Transaction::transactionFromSQL(
+                ledgerSeq, status, rawTxn, app);
+    }
+    catch (std::exception& e)
```

Comment:
this windows (travis) build is warning about `e` being unused here - you can just remove it

Reply:
..or add it to the log message..up to you

Reply:
Thanks for catching that!